nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -0,0 +1,1568 @@
+            .transaction_async(|conn| async move {
+                // Remove the record describing the collection itself.
+                let ncollections = {
+                    use db::schema::inv_collection::dsl;
+                    diesel::delete(
+                        dsl::inv_collection.filter(dsl::id.eq(collection_id)),
+                    )
+                    .execute_async(&conn)
+                    .await?;
+                };
+
+                // Remove rows for service processors.
+                let nsps = {
+                    use db::schema::inv_service_processor::dsl;
+                    diesel::delete(
+                        dsl::inv_service_processor
+                            .filter(dsl::inv_collection_id.eq(collection_id)),
+                    )
+                    .execute_async(&conn)
+                    .await?;
+                };
+
+                // Remove rows for roots of trust.
+                let nrots = {
+                    use db::schema::inv_root_of_trust::dsl;
+                    diesel::delete(
+                        dsl::inv_root_of_trust
+                            .filter(dsl::inv_collection_id.eq(collection_id)),
+                    )
+                    .execute_async(&conn)
+                    .await?;
+                };
+
+                // Remove rows for cabooses found.
+                let ncabooses = {
+                    use db::schema::inv_caboose::dsl;
+                    diesel::delete(
+                        dsl::inv_caboose
+                            .filter(dsl::inv_collection_id.eq(collection_id)),
+                    )
+                    .execute_async(&conn)
+                    .await?;
+                };
+
+                // Remove rows for errors encountered.
+                let nerrors = {
+                    use db::schema::inv_collection_error::dsl;
+                    diesel::delete(
+                        dsl::inv_collection_error
+                            .filter(dsl::inv_collection_id.eq(collection_id)),
+                    )
+                    .execute_async(&conn)
+                    .await?;
+                };
+
+                Ok((ncollections, nsps, nrots, ncabooses, nerrors))
+            })
+            .await
+            .map_err(|error| match error {
+                TransactionError::CustomError(e) => e,
+                TransactionError::Database(e) => {
+                    public_error_from_diesel(e, ErrorHandler::Server)
+                }
+            })?;
+
+        info!(&opctx.log, "removed inventory collection";
+            "collection_id" => collection_id.to_string(),
+            "ncollections" => ncollections,
+            "nsps" => nsps,
+            "nrots" => nrots,
+            "ncabooses" => ncabooses,
+            "nerrors" => nerrors,
+        );
+
+        Ok(())
+    }
+}
+
+/// A SQL common table expression (CTE) used to insert into `inv_caboose`
```

Comment:
I think this block comment is referencing code that no longer exists, right?

Reply:
Yikes!  Yes.  Removed in a55216d40.